Pdf/Model/Creditmemo.php

```diff
@@ -0,0 +1,128 @@
+
+class Creditmemo extends BaseCreditmemo
+{
+
+    /**
+     * @var \Magento\Framework\Filesystem\Directory\ReadInterface
+     */
+    protected $_rootDirectory;
+    /**
+     * @var Data
+     */
+    protected $_helper;
+
+
+    /**
+     * CreditmemoJp constructor.
+     * @param \Magento\Payment\Helper\Data $paymentData
+     * @param \Magento\Framework\Stdlib\StringUtils $string
+     * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+     * @param \Magento\Framework\Filesystem $filesystem
+     * @param \Magento\Sales\Model\Order\Pdf\Config $pdfConfig
+     * @param \Magento\Sales\Model\Order\Pdf\Total\Factory $pdfTotalFactory
+     * @param \Magento\Sales\Model\Order\Pdf\ItemsFactory $pdfItemsFactory
+     * @param \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate
+     * @param \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation
+     * @param \Magento\Sales\Model\Order\Address\Renderer $addressRenderer
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     * @param \Magento\Framework\Locale\ResolverInterface $localeResolver
+     * @param \MagentoJapan\Pdf\Helper\Data $_helper
+     * @param array $data
+     */
+    public function __construct(
+        \Magento\Payment\Helper\Data $paymentData,
+        \Magento\Framework\Stdlib\StringUtils $string,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
+        \Magento\Framework\Filesystem $filesystem,
+        \Magento\Sales\Model\Order\Pdf\Config $pdfConfig,
+        \Magento\Sales\Model\Order\Pdf\Total\Factory $pdfTotalFactory,
+        \Magento\Sales\Model\Order\Pdf\ItemsFactory $pdfItemsFactory,
+        \Magento\Framework\Stdlib\DateTime\TimezoneInterface $localeDate,
+        \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation,
+        \Magento\Sales\Model\Order\Address\Renderer $addressRenderer,
+        \Magento\Store\Model\StoreManagerInterface $storeManager,
+        \Magento\Framework\Locale\ResolverInterface $localeResolver,
+        Data $_helper,
+        array $data = []
+    )
+    {
+        $this->_rootDirectory = $filesystem->getDirectoryRead(DirectoryList::ROOT);
+        $this->_helper = $_helper;
+        parent::__construct(
+            $paymentData,
+            $string,
+            $scopeConfig,
+            $filesystem,
+            $pdfConfig,
+            $pdfTotalFactory,
+            $pdfItemsFactory,
+            $localeDate,
+            $inlineTranslation,
+            $addressRenderer,
+            $storeManager,
+            $localeResolver,
+            $data
+        );
+    }
+    /**
+     * @return Data
+     */
+    public function getHelper()
+    {
+        return $this->_helper;
+    }
+
+    /**
+     * @param $object
+     * @param int $size
+     * @return mixed
+     */
+    protected function _setFontRegular($object, $size = 7)
+    {
+        if ($this->getHelper()->getJapaneseFontIsActive()) {
```

Comment:
Such approach creates a lot of code duplication and if some 3rd party module has own pdf generated documents it will not be compatible.

We may try to create **after** plugin on `Magento\Framework\Filesystem\Directory\ReadInterface::getAbsolutePath` which will substitute result fort particular font paths. As we may limit check by `in_array` function call I believe there should not be dramatical performance degradation.

Reply:
I tried to set plugin against Magento\Framework\Filesystem\Directory\ReadInterface, but Magento\Framework\Filesystem\Directory\ReadFactory has hardcoded class name for generating Read class instance. Seems it's impossible to set plugin against ReadInterface.